ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java

```diff
@@ -3005,6 +3003,19 @@ private boolean validatePTFOperator(PTFOperator op, VectorizationContext vContex
           }
         }
       }
+      if (vectorPTFDesc.getOrderExprNodeDescs().length > 1) {
+        /*
+         * Currently, we need to rule out here all cases where a range boundary scanner can run,
+         * basically: 1. bounded start 2. bounded end which is not current row
+         */
+        if (windowFrameDef.getWindowType() == WindowType.RANGE
+            && (!windowFrameDef.isStartUnbounded() || !windowFrameDef.getEnd().isCurrentRow())) {
```

Comment:
I am not much aware of the range boundary scanner, but do we have any issue vectorizing UNBOUNDED FOLLOWING? Can you help me understand how does range boundary scanner affects the vectorization

Reply:
good catch, I'm going to fix the condition to enable unbounded end
FYI, the connection between this part and boundary scanner is the unimplemented method:
{code}
  @Override
  public boolean isDistanceGreater(Object v1, Object v2, int amt) {
    throw new UnsupportedOperationException("Only unbounded ranges supported");
  }
{code}

Reply:
this is fixed in https://github.com/apache/hive/pull/2099/commits/f6b5ea27501ccef175da961b49be9e0587a1ce00 and will be squashed later

Reply:
Thanks it looks good